Shuffling of player input.

BautistaDalmasso opened the pull request:
Messed around inputs are our bug!... I mean feature.

BautistaDalmasso committed:
Creates functions to shuffle the player input.

Messed around inputs are our bug!... I mean feature.

AbooMinister25 commented on server/mess_up_actions.py:
```diff
@@ -0,0 +1,59 @@
+from game_components.game_objects import Action, Player
+
+
+class messed_player:
```
Consider `MessedPlayer` for class names (CamelCase vs snake_case).

AbooMinister25 commented on server/mess_up_actions.py:
```diff
@@ -0,0 +1,59 @@
+        self.directions = _mess_up_directions()
+
+
+def _mess_up_directions() -> dict[str:str]:
```
the type hint should be `dict[str, str]`.

AbooMinister25 commented on server/mess_up_actions.py:
```diff
@@ -0,0 +1,59 @@
+    return dict(zip(directions, shuffled_directions))
+
+
+def _mess_up_actions(player: Player) -> dict[str:Action]:
```
Same as above, `dict[str, Action]`

AbooMinister25 commented on server/mess_up_actions.py:
```diff
@@ -0,0 +1,59 @@
+
+def shuffled_dict(dict_: dict) -> dict:
+    """Return a dict with it's keys and values shuffled around."""
+    shuffled_keys = list(dict_.keys())
```
You can just do `list(dict)`, no need for the `.keys()` call

BautistaDalmasso committed:
Sets up mechanism for sending actions.

Allows the client to communicate their actions and the server to respond
to those action requests.
TODO: Hook these requests to the actual game engine to execute actions.

it's too late to be coding so my mind probably didn't work very well
while writing this one :PPP

A5rocks committed:
Merge branch 'main' into mess-up-actions

A5rocks committed:
Fix types in the server

Now pyright doesn't complain about much except unimplemented functionality

A5rocks committed:
Move some common functionality into a shared package

A5rocks committed:
Move serialization into the common package

A5rocks committed:
Fix client type hints and make it use common schemas

BautistaDalmasso committed:
Reformat to use asyncronous for instead of recv in client.

BautistaDalmasso committed:
Fixes bug with UIDs not being generated uniquely.

The `data` passed to be hashed wasn't unique. Added the coordinates to
tis data to make it unique and prevent two rooms from having the same
UID.

AbooMinister25 reviewed (approved):
LGTM
